viz.go

```diff
@@ -0,0 +1,147 @@
+package dataset
+
+import (
+	"io"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ipfs/go-datastore"
+)
+
+// Viz stores configuration data related to representing a dataset as a
+// visualization
+type Viz struct {
+	// private storage for reference to this object
+	path datastore.Key
+	// Qri should always be "vc:0"
+	Qri Kind
+	// Format designates the visualization configuration syntax. currently the
+	// only supported syntax is "html"
+	Format string
+	// Script is a reader of raw script data
+	Script io.Reader `json:"_"`
+	// ScriptPath is the path to the script that created this
+	ScriptPath string `json:"script,omitempty"`
+}
+
+// Path gives the internal path reference for this structure
+func (v *Viz) Path() datastore.Key {
+	return v.path
+}
+
+// NewVizRef creates an empty struct with it's internal path set
+func NewVizRef(path datastore.Key) *Viz {
+	return &Viz{path: path}
+}
+
+// IsEmpty checks to see if Viz has any fields other than the internal path
+func (v *Viz) IsEmpty() bool {
+	return v.Format == "" && v.ScriptPath == ""
+}
+
+// SetPath sets the internal path property of a Viz
+// Use with caution. most callers should never need to call SetPath
+func (v *Viz) SetPath(path string) {
+	if path == "" {
+		v.path = datastore.Key{}
+	} else {
+		v.path = datastore.NewKey(path)
+	}
+}
+
+// Assign collapses all properties of a group of structures on to one this is
+// directly inspired by Javascript's Object.assign
+func (v *Viz) Assign(visConfigs ...*Viz) {
+	for _, vs := range visConfigs {
+		if vs == nil {
+			continue
+		}
+
+		if vs.path.String() != "" {
+			v.path = vs.path
+		}
+		if vs.Qri != "" {
+			v.Qri = vs.Qri
+		}
+		if vs.Format != "" {
+			v.Format = vs.Format
+		}
+		if vs.ScriptPath != "" {
+			v.ScriptPath = vs.ScriptPath
+		}
+	}
+}
+
+// _visconfig is a private struct for marshaling into & out of.
+// fields must remain sorted in lexographical order
+type _visconfig struct {
```

Comment:
Rename to vizconfigPod: we're no longer spelling viz with an "s", underscore unnecessary, this is playing the same role as DatasetPod, ProfilePod, etc.